hbase098/src/main/java/com/yahoo/ycsb/db/HBaseClient.java

```diff
@@ -22,576 +22,460 @@
-                            }
-                        }
-
-                        //System.out.println("Average latency: "+(((double)accum)/((double)opcount)));
-                        //System.out.println("Average get latency: "+(((double)cli.TotalGetTime)/((double)cli.TotalGetOps)));
-                    }
-                    catch (Exception e)
-                    {
-                        e.printStackTrace();
-                    }
-                }
-            };
-            allthreads.add(t);
-        }
+    final int threadcount = Integer.parseInt(args[0]);
 
-        long st=System.currentTimeMillis();
-        for (Thread t: allthreads)
-        {
-            t.start();
-        }
+    final String columnfamily = args[1];
 
-        for (Thread t: allthreads)
-        {
-            try
-            {
-                t.join();
-            }
-            catch (InterruptedException e)
-            {
+
+    final int opcount = Integer.parseInt(args[2]) / threadcount;
+
+    Vector<Thread> allthreads = new Vector<>();
+
+    for (int i = 0; i < threadcount; i++) {
+      Thread t = new Thread() {
+        public void run() {
+          try {
+            Random random = new Random();
+
+            HBaseClient cli = new HBaseClient();
+
+            Properties props = new Properties();
+            props.setProperty("columnfamily", columnfamily);
+            props.setProperty("debug", "true");
+            cli.setProperties(props);
+
+            cli.init();
+
+            long accum = 0;
+
+            for (int i = 0; i < opcount; i++) {
+              int keynum = random.nextInt(keyspace);
+              String key = "user" + keynum;
+              long st = System.currentTimeMillis();
+              Status result;
+              Vector<HashMap<String, ByteIterator>> scanResults = new Vector<>();
+              result = cli.scan("table1", "user2", 20, null, scanResults);
+
+              long en = System.currentTimeMillis();
+
+              accum += (en - st);
+
+              if (!result.equals(Status.OK)) {
+                System.out.println("Error " + result + " for " + key);
+              }
+
+              if (i % 10 == 0) {
+                System.out.println(i + " operations, average latency: " + (((double) accum) / ((double) i)));
+              }
             }
+          } catch (Exception e) {
+            e.printStackTrace();
+          }
         }
-        long en=System.currentTimeMillis();
-
-        System.out.println("Throughput: "+((1000.0)*(((double)(opcount*threadcount))/((double)(en-st))))+" ops/sec");
+      };
+      allthreads.add(t);
+    }
 
+    long st = System.currentTimeMillis();
+    for (Thread t : allthreads) {
+      t.start();
     }
-}
 
-/* For customized vim control
- * set autoindent
- * set si
- * set shiftwidth=4
-*/
+    for (Thread t : allthreads) {
+      try {
+        t.join();
+      } catch (InterruptedException ignored) {
```

Comment:
This is incorrect behavior. At a minimum the interrupted status for the current thread should be set.

It's a preexisting problem, not something from this fix, so I'd be fine with leaving it so long as there's an issue filed to fix it.